File: 2015/AI/AIClient_JAVA/src/event/EndGameEvent.java
/* ------------------------------------------------------------------------------
 ** _________   _________      ________    _____      _____  ___________ _________
 ** \_   ___ \ /   _____/     /  _____/   /  _  \    /     \ \_   _____//   _____/
 ** /    \  \/ \_____  \     /   \  ___  /  /_\  \  /  \ /  \ |    __)_ \_____  \ 
 ** \     \____/        \    \    \_\  \/    |    \/    Y    \|        \/        \
 **  \______  /_______  /     \______  /\____|__  /\____|__  /_______  /_______  /
 **         \/        \/             \/         \/         \/        \/        \/ 
 **
 ** EndGameEvent.java
 ** Event that happen when the game end
 **
 ** Author: Samuel-Ricardo Carriere
 ** ------------------------------------------------------------------------------*/

package event;

import world.World;

public class EndGameEvent extends IngoingEvent {

	@Override
	public void execute() {
		World.getInstance().endGame();
	}

	@Override
	public boolean fillArguments(String message) {
		return true;
	}
}

File: 2015/AI/AIClient_Python/src/event/EndGameEvent.py
'''
Created on Jan 24, 2015

@author: scarriere
'''
from event.IngoingEvent import IngoingEvent
from aiclient.Singleton import Singleton
from world.World import World

class EndGameEvent(IngoingEvent):

    def __init__(self):
        pass
    
    def fillArguments(self, string):
        pass
                
    def execute(self):
        world = Singleton(World)
        world._endGame()
        